rosetta/retriever/retriever_test.go

```diff
@@ -0,0 +1,1196 @@
+	})
+}
+
+func TestRetriever_Transaction(t *testing.T) {
+	testHeight := uint64(42)
+	testBlockID := identifier.Block{
+		Index: testHeight,
+		Hash:  "2c4c176c5c095bc3529ab425735077efb2afedd16c9ffc215a898df14fa8ac91",
+	}
+	testTransactionID := identifier.Transaction{
+		Hash: "0000000000000000000000000000000000000000000000000000000000000000",
+	}
+	depositType := &cadence.EventType{
+		Location:            utils.TestLocation,
+		QualifiedIdentifier: "deposit",
+		Fields: []cadence.Field{
+			{
+				Identifier: "amount",
+				Type:       cadence.UInt64Type{},
+			},
+			{
+				Identifier: "address",
+				Type:       cadence.AddressType{},
+			},
+		},
+	}
+	depositEvent := cadence.NewEvent(
+		[]cadence.Value{
+			cadence.NewUInt64(42),
+			cadence.NewAddress([8]byte{1, 2, 3, 4, 5, 6, 7, 8}),
+		},
+	).WithType(depositType)
+	depositEventPayload := json.MustEncode(depositEvent)
+
+	withdrawalType := &cadence.EventType{
+		Location:            utils.TestLocation,
+		QualifiedIdentifier: "withdrawal",
+		Fields: []cadence.Field{
+			{
+				Identifier: "amount",
+				Type:       cadence.UInt64Type{},
+			},
+			{
+				Identifier: "address",
+				Type:       cadence.AddressType{},
+			},
+		},
+	}
+	withdrawalEvent := cadence.NewEvent(
+		[]cadence.Value{
+			cadence.NewUInt64(42),
+			cadence.NewAddress([8]byte{2, 3, 4, 5, 6, 7, 8, 9}),
+		},
+	).WithType(withdrawalType)
+	withdrawalEventPayload := json.MustEncode(withdrawalEvent)
+
+	testEvents := []flow.Event{{
+		Type:    "deposit",
+		Payload: depositEventPayload,
+	}, {
+		Type:    "withdrawal",
+		Payload: withdrawalEventPayload,
+	}}
+
+	testTransaction := &object.Transaction{
+		ID: identifier.Transaction{
+			Hash: "0000000000000000000000000000000000000000000000000000000000000000",
+		},
+		Operations: []object.Operation{
+			{
+				ID:         identifier.Operation{},
+				RelatedIDs: []identifier.Operation(nil),
+				Type:       "TRANSFER",
+				Status:     "COMPLETED",
+				AccountID:  identifier.Account{Address: "0102030405060708"},
+				Amount: object.Amount{
+					Value:    "42",
+					Currency: identifier.Currency{Symbol: "FLOW", Decimals: 0x8},
+				},
+			},
+			{
+				ID:         identifier.Operation{},
+				RelatedIDs: []identifier.Operation(nil),
+				Type:       "TRANSFER",
+				Status:     "COMPLETED",
+				AccountID:  identifier.Account{Address: "0203040506070809"},
+				Amount: object.Amount{
+					Value: "42",
+					Currency: identifier.Currency{
+						Symbol: "FLOW", Decimals: 0x8,
+					},
+				},
+			},
+		},
+	}
+
+	t.Run("nominal case", func(t *testing.T) {
+		validator := &mocks.Validator{
+			BlockFunc:       func(block identifier.Block) (identifier.Block, error) { return block, nil },
+			TransactionFunc: func(transaction identifier.Transaction) error { return nil },
+		}
+		generator := &mocks.Generator{
+			TokensDepositedFunc: func(symbol string) (string, error) { return "", nil },
+			TokensWithdrawnFunc: func(symbol string) (string, error) { return "", nil },
```

Comment:
I wonder about the reasoning when you picked which mock functions to check the inputs on and which not? Wouldn't it make sense to be consistent and check them everywhere?